app/src/main/java/uk/gov/android/securestore/crypto/limitedmanager/LimitedCryptoManager.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.android.securestore.crypto.limitedmanager
+
+/**
+ * Interface to handle encryption and decryption of [String] data
+ */
+interface LimitedCryptoManager {
+    /**
+     * Encrypt a [String]
+     *
+     * @param input Plain text to encrypt
+     * @param callback Additional key for encryption
+     *
+     * @return Encrypted data as a [String]
+     *
+     * @throws [java.security.GeneralSecurityException] if encryption fails
+     */
+    fun encrypt(
+        input: String,
+        callback: (key: ByteArray) -> String?,
```

Comment:
I don't actually know what this is doing, is there something more helpful that we could call it?